packages/schematics/angular/utility/validation.ts

```diff
@@ -10,7 +10,7 @@ import { SchematicsException } from '@angular-devkit/schematics';
 
 // Must start with a letter, and must contain only alphanumeric characters or dashes.
 // When adding a dash the segment after the dash must also start with a letter.
-export const htmlSelectorRe = /^[a-zA-Z][.0-9a-zA-Z]*(:?-[a-zA-Z][.0-9a-zA-Z]*)*$/;
+export const htmlSelectorRe = /^[a-zA-Z][.0-9a-zA-Z]*(:?-[a-zA-Z][.0-9a-zA-Z]*(:?-[0-9]+)*)*$/;
```

Comment:
In some cases this regexp will fail to match

Example:
```
one-1
```

Maybe we can use the same logic as https://github.com/angular/angular-cli/blob/b66d94e5949102b2235b89cb75866973986ea30a/packages/angular_devkit/schematics/src/formats/html-selector.ts#L17C1-L44

//cc @clydin do you recall why we support upper characters as based on the spec it seems that these are not valid. https://html.spec.whatwg.org/multipage/custom-elements.html#valid-custom-element-name

Reply:
@alan-agius4 Found another problem. If we allow `one-1` like selectors then it will also verify `app-1-one` as valid and in that case, the component class name becomes like this: -

```
export class 1OneComponent {
}
```

This is invalid class name. Isn't it?

So, I might have to add some conditions that regexp doesn't allow `1-one` if the component is being created with a prefix which will ultimately become `app-1-one`.